### PR TITLE
ci: update release ci yml

### PR DESCRIPTION
**Pull request by yuluo-yx**
- For https://github.com/LinuxSuRen/api-testing/pull/414#issuecomment-2101818968

### CODE REVIEW

**Comment by yuluo-yx** [comment]
ptal @LinuxSuRen

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -40,13 +36,10 @@ jobs:
...
-      - uses: actions/checkout@v4
-      - name: Set up Go
-        uses: actions/setup-go@v4
-        with:
-          go-version: 1.22.2
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b  # v4.1.4
+      - uses: ./tools/github-actions/setup-deps
       - name: Unit Test
         run: |
           make test-all-backend
```
Comment (LinuxSuRen) [inline .github/workflows/release.yaml:45]: Forgot?

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Need to update https://github.com/LinuxSuRen/api-testing/pull/421/files#diff-e426ed45842837026e10e66af23d9c7077e89eacbe6958ce7cb991130ad05adaL125

File: .github/workflows/release.yaml (hunk 1 above)
Comment (yuluo-yx) [reply .github/workflows/release.yaml:45]: sorry 😢

Comment (yuluo-yx) [reply .github/workflows/release.yaml]: done!

**Review by LinuxSuRen — APPROVED** [review]

**Pull request merged**